tests/tests/test-precompile/test-precompile-proxy.ts

```diff
@@ -0,0 +1,247 @@
+import "@polkadot/api-augment";
+import "@moonbeam-network/api-augment";
+import { expect } from "chai";
+import { ALITH_ADDRESS, BALTATHAR_ADDRESS, CHARLETH_ADDRESS } from "../../util/accounts";
+
+import { describeDevMoonbeam } from "../../util/setup-dev-tests";
+import { getCompiled } from "../../util/contracts";
+import { ethers } from "ethers";
+import { ALITH_TRANSACTION_TEMPLATE, createTransaction } from "../../util/transactions";
+import {
+  CONTRACT_PROXY_TYPE_GOVERNANCE,
+  CONTRACT_PROXY_TYPE_STAKING,
+  PRECOMPILE_PROXY_ADDRESS,
+} from "../../util/constants";
+import { expectEVMResult } from "../../util/eth-transactions";
+
+const PROXY_CONTRACT_JSON = getCompiled("Proxy");
+const PROXY_INTERFACE = new ethers.utils.Interface(PROXY_CONTRACT_JSON.contract.abi);
+
+describeDevMoonbeam("Precompile - Proxy - add proxy fails if pre-existing proxy", (context) => {
+  before("add proxy account", async () => {
+    const { result } = await context.createBlock(
+      createTransaction(context, {
+        ...ALITH_TRANSACTION_TEMPLATE,
+        to: PRECOMPILE_PROXY_ADDRESS,
+        data: PROXY_INTERFACE.encodeFunctionData("addProxy", [
+          BALTATHAR_ADDRESS,
+          CONTRACT_PROXY_TYPE_STAKING,
```

Comment:
You might be able to create the polkadot type and then do a .tou8a(). This way you dont have to store those contracts. But up to you

Reply:
@crystalin do we prefer named constants or `createType`?

```typescript
CONTRACT_PROXY_TYPE_STAKING;

// or

context.polkadotApi.createType("ProxyType", "Staking").toU8a(); 
```